### PR TITLE
4 add set detection area endpoint

### PR DESCRIPTION
**Pull request by ao-picterra**

### CODE REVIEW

File: src/Client.js [hunk 1]
```diff
@@ -194,6 +195,71 @@ export default class APIClient {
...
+      while (!isReady) // Poll until complete
+      // Error management
+      if (!isReady) {
+        const text = await response.text()
+        throw new APIError('Error uploading detection area', text)
+      }
+      // Acknowledge all went good
+      return true
+    } catch (error) {
+      throw new APIError(error)
```
Comment (julienr) [inline src/Client.js:260]: I wouldn't have this global try/catch here. I feel making all error seem like APIError isn't super useful for the end-user: as a user, I'd rather have the choice to catch all myself or catch specific individual exceptions (which isn't possible anymore if we wrap)

**Review by julienr — COMMENTED** [review]
Cool, LGTM overall

File: src/Client.js [hunk 2]
```diff
@@ -194,6 +195,71 @@ export default class APIClient {
...
+     * @summary Set the detection area of an available raster
+     * @description Given a raster, sets the detection area geometries for it, overriding
+     * any previous one
+     * @param {String} rasterId The Id of the raster whose detection area we want to set
+     * @param {String} fileName The GeoJSON with the Detection Areas geometries
+     * @returns {Promise<Boolean>} Whether or not the operation succedeed
+     * @throws {APIError} Containing error code and text
+     * @example TODO
+     */
+  async setRasterDetectionArea (fileName, rasterId) {
```
[the remark this answers is not in the document]
Comment (ao-picterra) [reply src/Client.js]: Checked

File: src/Client.js (hunk 1 above)
Comment (ao-picterra) [reply src/Client.js]: Checked

**Review by julienr — APPROVED** [review]

**Pull request merged**